backend/cfg/cfg_regalloc_utils.ml

```diff
@@ -456,3 +461,68 @@ let may_use_stack_operands_everywhere :
   may_use_stack_operands_array map instr.arg;
   may_use_stack_operands_array map instr.res;
   All_spilled_registers_rewritten
+
+let insert_spill_block :
+    Cfg_with_layout.t ->
+    Instruction.t list ->
+    after:Cfg.basic_block ->
+    terminator_id:int ->
+    unit =
+ fun cfg_with_layout spills ~after:predecessor_block ~terminator_id ->
+  let cfg = Cfg_with_layout.cfg cfg_with_layout in
+  let successors =
+    Cfg.successor_labels ~normal:true ~exn:false predecessor_block
+  in
+  let successor_label =
+    match Label.Set.to_seq successors |> List.of_seq with
+    | [label] -> label
+    | [] ->
+      Misc.fatal_errorf
+        "Cannot insert a block after block %a: it has no successors" Label.print
+        predecessor_block.start
+    | _ :: _ :: _ ->
+      Misc.fatal_errorf
+        "Cannot insert a block after block %a: it has more than one successor"
+        Label.print predecessor_block.start
```

Comment:
(I am not sure about this restriction.)

Reply:
The weakest restriction would be that the set of successors is of size 1.
I believe it could be more confusing that helpful.

Alternatively we could consider passing both the predecessor and the successor to this function, basically we would be adding a block on a specific edge and not just "after" another block.

Reply:
I think my point is that one should be
able to insert spills even in the presence
in multiple successors. I have even seen
CFG designs were the spill instructions
were "along" the edges (which is not
totally crazy for some later optimizations
such as merging in-line blocks, simplifying
cases were multiple successors are actually
the same destination, etc).

Since multiple successors when inserting
spills cannot happen as of today because
of the operations we have, I am fine with
a CR. I just don't want the code to convey
the idea that it is an impossible case in
general.

Reply:
Yes, now it inserts along the edges to all the successors, copying the instructions if necessary.